pkg/controller/logstorage/users/users_controller.go

```diff
@@ -357,15 +358,51 @@ func (r *UsersCleanupController) Reconcile(ctx context.Context, request reconcil
 		request.Namespace, "Request.Name", request.Name, "installNS", helper.InstallNamespace(), "truthNS", helper.TruthNamespace())
 	reqLogger.Info("Reconciling LogStorage - Cleanup")
 
-	// Wait for Elasticsearch to be installed and available.
-	elasticsearch, err := utils.GetElasticsearch(ctx, r.client)
-	if err != nil {
+	// We skip requests without a namespace specified in multi-tenant setups.
+	if r.multiTenant && request.Namespace == "" {
+		return reconcile.Result{}, nil
+	}
+
+	// Check if this is a tenant-scoped request.
+	_, _, err := utils.GetTenant(ctx, r.multiTenant, r.client, request.Namespace)
+	if errors.IsNotFound(err) {
+		reqLogger.Info("No Tenant in this Namespace, skip")
+		return reconcile.Result{}, nil
+	} else if err != nil {
 		return reconcile.Result{}, err
 	}
-	if elasticsearch == nil || elasticsearch.Status.Phase != esv1.ElasticsearchReadyPhase {
+
+	// Get LogStorage resource.
+	logStorage := &operatorv1.LogStorage{}
+	err = r.client.Get(ctx, utils.DefaultTSEEInstanceKey, logStorage)
+	if err != nil {
```

Comment:
Just to double check, this change isn't for the current bug in cc, this is something else you noticed, right?

I think in general we try to do this CR check first since there's not really any point in doing anything else if the CR doesn't exist.